### PR TITLE
fix(#3366): add detailed observability logs to AutoFixer _get_changed_files

### PR DESCRIPTION
**Pull request by RC918**
## Summary

Adds comprehensive observability logging to the `_get_changed_files` method in AutoFixer to help diagnose why CI failure auto-fix attempts fail in staging.

During testing of the CI failure fast path (Issue #3366), we discovered that AutoFixer was being triggered correctly but failing with "No changed files found". The existing logs only showed:
- `[AutoFixer] github_api not available, using git diff`
- `[AutoFixer] No changed files found, skipping auto-fix`

This PR adds detailed logging at each step:
- Method entry with pr_number, repo, trace_id
- GitHub API import success/failure with specific error details
- Git diff fallback with cwd, returncode, stdout_len, stderr
- File count and first 5 files for both methods
- Final warning when returning empty list

All logs include `trace_id` for correlation with the full request flow.

## Updates since last revision

Addressed code review feedback:
- **Dynamic base ref detection**: Instead of hardcoding `origin/main`, now attempts to detect the default branch via `git symbolic-ref refs/remotes/origin/HEAD` (falls back to `origin/main` if detection fails)
- **Improved stderr handling**: Increased preview limit to 500 chars, added `stderr_truncated` flag, and logs stderr as a separate warning when present
- **Structured logging improvements**: Added `base_ref` and `files_preview` to log extra dict for better structured logging

## Review & Testing Checklist for Human

- [ ] Verify log messages don't expose sensitive information (they log file paths and error messages, but no credentials)
- [ ] Deploy to staging and trigger a CI failure to verify the new logs appear in Render worker logs
- [ ] Search for `[AutoFixer] _get_changed_files` in logs to see the complete diagnostic output

### Notes

This is a diagnostic-only change with no functional modifications. The root cause of the AutoFixer failure is that:
1. `github_api` import fails (ImportError) in the worker environment
2. `git diff` fallback returns empty because there's no git repository in the Render worker

A follow-up fix will be needed to address the actual root cause (likely need to pass changed files from the webhook payload or use GitHub API with proper initialization).

**Link to Devin run:** https://app.devin.ai/sessions/9bab8bd3f6904a1db27deca6fd525ac4
**Requested by:** Ryan Chen (@RC918)

### CODE REVIEW

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review

### File-Level Comments

*The following comments could not be posted inline:*

**`handoff/20250928/40_App/orchestrator/project_engineer/fixer_integration.py`**
> [Line info removed: line_433_is_context_not_addition] IMPACT: Hardcoding 'origin/main' may fail in repositories with non-standard remote names or default branches (e.g., 'main' vs 'master', or upstream forks). SUGGESTED FIX: Derive base branch dynamically via settings or git config to improve robustness across environments.

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:build] No files were found with the provided path: packages/shared-ui/coverage/. No artifacts will be uploaded.

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:build] No files were found with the provided path: handoff/20250928/40_App/owner-console/coverage/. No artifacts will be uploaded.

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:build] No files were found with the provided path: handoff/20250928/40_App/frontend-dashboard/coverage/. No artifacts will be uploaded.

**`handoff/20250928/40_App/frontend-dashboard/src/components/Dashboard.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useCallback has a missing dependency: 'setDashboardLayout'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/Dashboard.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useCallback has a missing dependency: 'setDashboardLayout'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/CostAnalysis.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useEffect has a missing dependency: 'loadCostData'. Either include it or remove the dependency array

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useCallback has a missing dependency: 'getFeaturesByPlan'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/AgentGovernance.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element

**`handoff/20250928/40_App/frontend-dashboard/src/components/AgentGovernance.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] Visible, non-interactive elements with click handlers must have at least one keyboard listener                                                                                                   

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/__tests__/TotpInput.test.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] The autoFocus prop should not be used, as it can reduce usability and accessibility for users

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/__tests__/TotpInput.test.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] The autoFocus prop should not be used, as it can reduce usability and accessibility for users

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/TwoFactorEnroll.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] The autoFocus prop should not be used, as it can reduce usability and accessibility for users                                          

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/TwoFactorEnroll.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useEffect has missing dependencies: 'enrollData' and 'handleEnrollStart'. Either include them or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/Dashboard.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useCallback has a missing dependency: 'setDashboardLayout'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/Dashboard.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useCallback has a missing dependency: 'setDashboardLayout'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/CostAnalysis.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useEffect has a missing dependency: 'loadCostData'. Either include it or remove the dependency array

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useCallback has a missing dependency: 'getFeaturesByPlan'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/AgentGovernance.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element

**`handoff/20250928/40_App/frontend-dashboard/src/components/AgentGovernance.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] Visible, non-interactive elements with click handlers must have at least one keyboard listener                                                                                                   

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/__tests__/TotpInput.test.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] The autoFocus prop should not be used, as it can reduce usability and accessibility for users

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/__tests__/TotpInput.test.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] The autoFocus prop should not be used, as it can reduce usability and accessibility for users

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/TwoFactorEnroll.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] The autoFocus prop should not be used, as it can reduce usability and accessibility for users                                          

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/TwoFactorEnroll.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useEffect has missing dependencies: 'enrollData' and 'handleEnrollStart'. Either include them or remove the dependency array

**`handoff/20250928/40_App/owner-console/src/components/2fa/RegenerateBackupCodesModal.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "bg-yellow-50" detected. Use semantic design tokens instead: bg-warning-50      

**`handoff/20250928/40_App/owner-console/src/components/2fa/DisableTwoFAModal.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "text-yellow-800" detected. Use semantic design tokens instead: text-warning-800

**`handoff/20250928/40_App/owner-console/src/components/2fa/DisableTwoFAModal.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "text-yellow-600" detected. Use semantic design tokens instead: text-warning-600

**`handoff/20250928/40_App/owner-console/src/components/2fa/DisableTwoFAModal.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "bg-yellow-50" detected. Use semantic design tokens instead: bg-warning-50      

**`handoff/20250928/40_App/owner-console/src/components/2fa/BackupCodesList.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "text-yellow-800" detected. Use semantic design tokens instead: text-warning-800    

**`handoff/20250928/40_App/owner-console/src/components/2fa/BackupCodesList.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "bg-yellow-50" detected. Use semantic design tokens instead: bg-warning-50          

**`handoff/20250928/40_App/owner-console/src/components/2fa/BackupCodesList.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "text-yellow-800" detected. Use semantic design tokens instead: text-warning-800    

**`handoff/20250928/40_App/owner-console/src/components/2fa/BackupCodesList.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "border-yellow-200" detected. Use semantic design tokens instead: border-warning-200

**`handoff/20250928/40_App/owner-console/src/App.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "border-purple-600" detected. Use semantic design tokens instead: border-accent-600

**`handoff/20250928/40_App/owner-console/.storybook/preview.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Fast refresh only works when a file only exports components. Move your component(s) to a separate file

File: handoff/20250928/40_App/orchestrator/project_engineer/fixer_integration.py [hunk 1]
```diff
@@ -354,33 +355,174 @@ async def _get_changed_files(self, state: Dict[str, Any]) -> List[str]:
...
                 timeout=30
             )
+
+            # Handle stderr with truncation indicator
+            stderr_preview = ""
+            stderr_truncated = False
+            if result.stderr:
+                if len(result.stderr) > 500:
+                    stderr_preview = result.stderr[:500]
+                    stderr_truncated = True
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/project_engineer/fixer_integration.py]: IMPACT: Hardcoding 500 as the truncation limit without a constant or config makes it harder to adjust globally or test edge cases. This magic number could lead to inconsistent log behavior if reused elsewhere. SUGGESTED FIX: Define MAX_STDERR_PREVIEW_LENGTH = 500 at module level or in settings to make it configurable and self-documenting.

File: handoff/20250928/40_App/orchestrator/project_engineer/fixer_integration.py [hunk 2]
```diff
@@ -354,33 +355,174 @@ async def _get_changed_files(self, state: Dict[str, Any]) -> List[str]:
         Returns:
             List of file paths
         """
+        trace_id = state.get("trace_id", "unknown")
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/project_engineer/fixer_integration.py]: IMPACT: Using 'unknown' as default for trace_id may complicate log correlation if tracing context is missing. It could mask initialization issues. SUGGESTED FIX: Generate a synthetic trace ID (e.g., 'autofixer-fallback-' + timestamp) when not provided to maintain end-to-end traceability.

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review

### File-Level Comments

*The following comments could not be posted inline:*

**`handoff/20250928/40_App/frontend-dashboard/src/components/Dashboard.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useCallback has a missing dependency: 'setDashboardLayout'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/Dashboard.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useCallback has a missing dependency: 'setDashboardLayout'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/CostAnalysis.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useEffect has a missing dependency: 'loadCostData'. Either include it or remove the dependency array

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useCallback has a missing dependency: 'getFeaturesByPlan'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/AgentGovernance.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element

**`handoff/20250928/40_App/frontend-dashboard/src/components/AgentGovernance.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] Visible, non-interactive elements with click handlers must have at least one keyboard listener                                                                                                   

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/__tests__/TotpInput.test.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] The autoFocus prop should not be used, as it can reduce usability and accessibility for users

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/__tests__/TotpInput.test.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] The autoFocus prop should not be used, as it can reduce usability and accessibility for users

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/TwoFactorEnroll.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] The autoFocus prop should not be used, as it can reduce usability and accessibility for users                                          

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/TwoFactorEnroll.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard Tests] React Hook useEffect has missing dependencies: 'enrollData' and 'handleEnrollStart'. Either include them or remove the dependency array

**`handoff/20250928/40_App/owner-console/src/components/2fa/RegenerateBackupCodesModal.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "bg-yellow-50" detected. Use semantic design tokens instead: bg-warning-50      

**`handoff/20250928/40_App/owner-console/src/components/2fa/DisableTwoFAModal.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "text-yellow-800" detected. Use semantic design tokens instead: text-warning-800

**`handoff/20250928/40_App/owner-console/src/components/2fa/DisableTwoFAModal.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "text-yellow-600" detected. Use semantic design tokens instead: text-warning-600

**`handoff/20250928/40_App/owner-console/src/components/2fa/DisableTwoFAModal.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "bg-yellow-50" detected. Use semantic design tokens instead: bg-warning-50      

**`handoff/20250928/40_App/owner-console/src/components/2fa/BackupCodesList.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "text-yellow-800" detected. Use semantic design tokens instead: text-warning-800    

**`handoff/20250928/40_App/owner-console/src/components/2fa/BackupCodesList.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "bg-yellow-50" detected. Use semantic design tokens instead: bg-warning-50          

**`handoff/20250928/40_App/owner-console/src/components/2fa/BackupCodesList.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "text-yellow-800" detected. Use semantic design tokens instead: text-warning-800    

**`handoff/20250928/40_App/owner-console/src/components/2fa/BackupCodesList.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "border-yellow-200" detected. Use semantic design tokens instead: border-warning-200

**`handoff/20250928/40_App/owner-console/src/App.jsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Hardcoded color class "border-purple-600" detected. Use semantic design tokens instead: border-accent-600

**`handoff/20250928/40_App/owner-console/.storybook/preview.tsx`**
> [Line info removed: file_not_in_diff] [check_run:Owner Console Tests] Fast refresh only works when a file only exports components. Move your component(s) to a separate file

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:Frontend Dashboard E2E Tests (Design Tokens)] No files were found with the provided path: handoff/20250928/40_App/frontend-dashboard/test-results/. No artifacts will be uploaded.

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:build] No files were found with the provided path: packages/shared-ui/coverage/. No artifacts will be uploaded.

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:build] No files were found with the provided path: handoff/20250928/40_App/owner-console/coverage/. No artifacts will be uploaded.

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:build] No files were found with the provided path: handoff/20250928/40_App/frontend-dashboard/coverage/. No artifacts will be uploaded.

**`handoff/20250928/40_App/frontend-dashboard/src/components/Dashboard.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useCallback has a missing dependency: 'setDashboardLayout'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/Dashboard.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useCallback has a missing dependency: 'setDashboardLayout'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/CostAnalysis.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useEffect has a missing dependency: 'loadCostData'. Either include it or remove the dependency array

**`.github`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useCallback has a missing dependency: 'getFeaturesByPlan'. Either include it or remove the dependency array

**`handoff/20250928/40_App/frontend-dashboard/src/components/AgentGovernance.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element

**`handoff/20250928/40_App/frontend-dashboard/src/components/AgentGovernance.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] Visible, non-interactive elements with click handlers must have at least one keyboard listener                                                                                                   

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/__tests__/TotpInput.test.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] The autoFocus prop should not be used, as it can reduce usability and accessibility for users

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/__tests__/TotpInput.test.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] The autoFocus prop should not be used, as it can reduce usability and accessibility for users

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/TwoFactorEnroll.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] The autoFocus prop should not be used, as it can reduce usability and accessibility for users                                          

**`handoff/20250928/40_App/frontend-dashboard/src/components/2fa/TwoFactorEnroll.tsx`**
> [Line info removed: file_not_in_diff] [check_run:build] React Hook useEffect has missing dependencies: 'enrollData' and 'handleEnrollStart'. Either include them or remove the dependency array
